Add documentation for ocaml-tiff library

pritgit28 opened the pull request:
@patricoferris Hello! I have written a documentation based on my understanding of the codebase in this repository. Kindly review the content and let me know if there are any changes or additions you would recommend. 
Thank you!

patricoferris commented on documentation.md:
```diff
@@ -0,0 +1,114 @@
+## Test cases
+
+### Test coverage
+The project includes a suite of unit tests that validate the following aspects of the TIFF decoding pipeline:
```
This is also not needed.

patricoferris commented:
No worries @pritgit28 -- just trying to understand where you are coming from with this change. Glad to see you have a really great understanding of the TIFF format.

patricoferris commented on documentation.md:
```diff
@@ -0,0 +1,50 @@
+
+## Main file responsibilities in ocaml-tiff
+
+### tiff.ml 
```
A bit like the tests, I think this kind of verbosity in the markdown file is hard to keep in sync (I see a #41 PR which might already make this outdated if it were to be merged). How about we shift these descriptions into comments in the head of the  `.ml` files?

pritgit28 commented:
@patricoferris Hi! I've added the file descriptions on the head of each file. And as I edited the initial markdown file I presented, the contents that were left out felt like they would make sense in a HACKING.md file so I renamed the file with the contents present now. Please review what I've pushed and I will make the changes mentioned in the review comments.

Thank you :)

pritgit28 commented:
@patricoferris Can you please help me with how this pr and other two issues I opened will be dealt?
